pkg/pprof/fix_go_truncated.go

```diff
@@ -23,11 +23,20 @@ const (
 // MayHaveGoHeapTruncatedStacktraces reports whether there are
 // any chances that the profile may have truncated stack traces.
 func MayHaveGoHeapTruncatedStacktraces(p *profilev1.Profile) bool {
-	if !hasGoHeapSampleTypes(p) {
+	var minDepth int
+
+	if hasGoHeapSampleTypes(p) {
+		// TODO: Go heap profiles in Go 1.23 have a depth limit of 128 frames. At some point we can probably increase/remove this
+		minDepth = 32
+	} else if hasGoCPUSampleTypes(p) {
+		minDepth = 64
+	} else {
 		return false
 	}
-	// Some truncated stacks have depth less than the depth limit (32).
-	const minDepth = 28
+
+	// Some truncated stacks have depth less than the depth limit.
+	minDepth -= 3
+
```

Comment:
I'm wondering why we set it to 3. It was set to max-4 because in Go mem profiler they do cut 4 frames:
https://github.com/golang/go/blob/f7c330eac7777612574d8a1652fd415391f6095e/src/runtime/mprof.go#L446

Reply:
Yeah sorry this was me testing something and accidentally comitting it.